src/pages/on-chain-governance/_components/ConfirmVoteDialog.tsx

```diff
@@ -171,166 +153,236 @@ export function ConfirmVoteDialog({
+
   return (
     <>
       <div className="mt-2 mb-6">
-        <span className="text-lg text-gray-600 dark:text-gray-100">
+        <span
+          data-testid="OnChainGovernance.VotingFlow.VoteForProposal.SubTitle"
+          className="text-lg text-gray-600 dark:text-gray-100"
+        >
           Confirm your masternode and vote on the proposal
         </span>
       </div>
-      <div className="mb-4">
-        <span className="text-gray-600 dark:text-gray-100 font-semibold text-sm">
-          Masternode
-        </span>
-        <div className="flex flex-row items-center gap-x-2 mt-1">
-          <div
-            onBlur={() => {
-              setIsMasterNodeInputFocus(false);
-            }}
-            onFocus={() => {
-              setIsMasterNodeInputFocus(true);
+
+      <span
+        data-testid="OnChainGovernance.VotingFlow.VoteForProposal.Masternode"
+        className="text-gray-600 dark:text-gray-100 font-semibold text-sm tracking-[0.0025em]"
+      >
+        Masternode
+      </span>
+
+      <div className="flex flex-row items-center gap-x-2 mt-1">
+        <div
+          data-testid="OnChainGovernance.VotingFlow.VoteForProposal.MasternodeInput"
+          onBlur={() => {
+            setIsMasterNodeInputFocus(false);
+          }}
+          onFocus={() => {
+            setIsMasterNodeInputFocus(true);
+          }}
+          className={classNames(
+            "flex flex-row rounded border border-gray-300 py-3 px-4 w-full dark:bg-gray-800",
+            { "border-primary-300": isMasterNodeInputFocus },
+            { "border-red-500": masterNodeErrorMsg !== "" }
+          )}
+        >
+          <textarea
+            ref={ref}
+            onChange={(v) => {
+              if (v.target.value.length !== 64) {
+                setMasterNodeErrorMsg("Please enter a valid masternode ID");
+              } else {
+                setMasterNodeErrorMsg("");
+              }
+              setMasterNodeID(v.target.value);
             }}
-            className={classNames(
-              "flex flex-row rounded border py-3 px-4 w-full dark:bg-gray-800",
-              { "border-primary-300": isMasterNodeInputFocus },
-              { "border-red-600": masterNodeErrorMsg !== "" }
-            )}
-          >
-            <textarea
-              onChange={(v) => {
-                if (v.target.value.length !== 64) {
-                  setMasterNodeErrorMsg("Enter a valid masternode ID");
-                } else {
-                  setMasterNodeErrorMsg("");
-                }
-                setMasterNodeID(v.target.value);
+            value={masterNodeID}
+            className="overflow-visible resize-none text-sm focus:outline-none focus:caret-[#007AFF] dark:bg-gray-800 dark:text-dark-gray-900 disabled:bg-white dark:disabled:bg-gray-800 placeholder:text-gray-400 grow tracking-[0.0025em]"
+            placeholder="Masternode ID"
+          />
+
+          {isMasterNodeInputFocus && (
+            <MdClear
```

Comment:
can we use this clear btn here?
![image](https://user-images.githubusercontent.com/53080940/211245843-32b99d92-2010-425a-a965-c0c73e291231.png)

Reply:
can we add margin left 4px for implemented clr btn 
<img width="547" alt="image" src="https://user-images.githubusercontent.com/53080940/211484731-6d1e3208-c366-4b0f-bc13-a4dbb885d4af.png">

Reply:
updated to fix width of the input text and use justify so that text would not jump
<img width="444" alt="image" src="https://user-images.githubusercontent.com/57183851/211491619-aeb7f8a2-d239-432b-a80f-9a2a13ed6c06.png">